// 3.0.0/src/main/resources/static/app/app.module.ts
import { NgModule } from '@angular/core';


import { BrowserModule } from '@angular/platform-browser';


/* App Root */
import { AppComponent } from './app.component';

/* Routing Module */
import { AppRoutingModule } from './app-routing.module';

/*The modules to be loaded with App Root */
//import { CustomerActionModule } from "./customeractionsmodule/customeractions.module";
//import { PhonemaintComponent } from "./phonemaintmodule/phonemaint.component";
//import { EmailmaintComponent } from "./emailmaintmodule/emailmaint.component";
import { FormsModule } from "@angular/forms";
//import { CustomerService } from "./shared/services/customer.service";
//import { CustomerActionsComponent } from "./customeractionsmodule/customeractions.component";
//import { PhoneMaintModule } from "./phonemaintmodule/phonemaint.module";
//import { EmailMaintModule } from "./emailmaintmodule/emailmaint.module";
//import { PhonemaintService } from "./phonemaintmodule/phonemaint.service";
//import { Logger } from "./shared/services/logger.service";
import { HttpModule } from '@angular/http';
import { CustomerService } from "./shared/services/customer.service";
import { PhonemaintService } from "./phonemaintmodule/phonemaint.service";
import { SessionStorageService } from 'ngx-webstorage';
//import {CookieService} from 'angular2-cookie/core';
//import { CookieModule, CookieService } from 'ngx-cookie';




@NgModule( {
    exports: [],
    declarations: [AppComponent],
    imports: [
              BrowserModule,
              //CustomerActionModule,
              HttpModule,
              FormsModule,
              AppRoutingModule
       
        
    ],
    providers: [
    CustomerService,PhonemaintService,SessionStorageService 
      
    ],
    bootstrap: [AppComponent]
} )
export class AppModule { }
