// 3.0.0/src/main/resources/static/app/emailmaintmodule/emailmaint.component.ts
import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { Title } from '@angular/platform-browser';
import { NgForm } from '@angular/forms';
import { Customer } from "../shared/models/customer";
import { CustomerService } from "../shared/services/customer.service";



@Component({
    moduleId: module.id,
    selector: 'atlas-email',
    templateUrl: 'emailmaint.component.html',
    styleUrls: ['emailmaint.component.css']
  })
  
export class EmailmaintComponent {
    customer: Customer;  
    model:any={};
    mailmaintanceload : any;
    mailchannelradio:string;
    transfertobfu:boolean = false;
    constructor(
            private titleService: Title,
            private customerService: CustomerService,private router: Router) {
    
        titleService.setTitle('ATLAS - Phone Maintenance');
    }
     
    ngOnInit() { 
        this.model.mailmaintanceload="Please Select";
        this.customer = this.customerService.getCustomer();
    }
    mailReset(){
        this.transfertobfu = true;      
    }
    mailByPass(forwardUrl: string ){
        
        this.router.navigate( [forwardUrl] );
    }
    
}
