// 3.0.0/src/main/resources/static/app/accountactionsmodule/accountactions.component.ts
import { Component,OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { Title } from '@angular/platform-browser';
import { CustomerService } from "../shared/services/customer.service";
declare let customerLinksObject: any;


@Component( {
    moduleId: module.id,
    selector: 'atlas-accountactions',
    templateUrl: 'accountactions.component.html',
    styleUrls: ['accountactions.component.css']
} )

export class AccountActionsComponent implements OnInit {
       accountactionlinksreq: any={"buttongroupname": "Account Actions","buttongrouplob": "DEP"};
       accountractionlinks: any;
       accountVO: string;
       accountactionlinksbuttongroupwebmigrated: any;
       accountactionlinkscommunication: any;
       accountactionlinkscommunicationactiongroup: any;
        constructor( private router: Router, private customerService: CustomerService, private titleService: Title ) {

        titleService.setTitle( 'ATLAS - Customer Actions' );
    }
    
    ngOnInit() {
            this.loadAccountLinks(this.accountactionlinksreq);     
    }
    
    loadAccountLinks(accountactionlinksreq: any){
        this.customerService.retrieveCustomerLinks(accountactionlinksreq)
            .subscribe(
                    accountractionlinks => {
                    this.accountractionlinks = accountractionlinks;
                    console.log('accountractionlinks'+JSON.stringify(this.accountractionlinks));
                    this.accountactionlinksbuttongroupwebmigrated = this.accountractionlinks.buttongroupwebmigrated;
                    if(this.accountractionlinks.buttongroupwebmigrated == true){
                        this.accountactionlinkscommunication = accountractionlinks.buttonGroupMap.Communication;
                        this.accountactionlinkscommunicationactiongroup = this.accountactionlinkscommunication[0].actiongroup;
                        
                        /*this.customeractionlinksprofile = customeractionlinks.buttonGroupMap.Profile;
                        this.customeractionlinksprofileactiongroup = this.customeractionlinksprofile[0].actiongroup;
                        this.customeractionlinkscommunication = customeractionlinks.buttonGroupMap.Communication;
                        this.customeractionlinkscommunicationactiongroup = this.customeractionlinkscommunication[0].actiongroup;
                        this.customeractionlinksOnBoarding = customeractionlinks.buttonGroupMap.Onboarding;
                        this.customeractionlinksOnBoardingactiongroup = this.customeractionlinksOnBoarding[0].actiongroup;*/
                    }
                    else if(this.accountactionlinksbuttongroupwebmigrated == false){
                          //window.close();
                          customerLinksObject.invokeCustomerActions(this.accountractionlinks.buttongroupname,this.accountractionlinks.buttongroupclassname);
                    }
                    },
                    error => {
                        console.log(error);
                        //this.loading = false;
                    });
    }
    
    updateModelAndForward( forwardUrl: any ) {
       this.accountVO = ( <HTMLInputElement>document.getElementById( 'accountVO' ) ).value;
        this.accountVO = this.accountVO.replace( /'/g, "" );

        this.customerService.setCustomer(JSON.parse(this.accountVO));
        
    }
     AccountActionLinks(linkname:string,linkclassname:string) {
    		customerLinksObject.invokeCustomerActions(linkname,linkclassname);
  	}
    
}
