// 3.0.0/rollup-config.js
import rollup      from 'rollup';
import nodeResolve from 'rollup-plugin-node-resolve';
import commonjs    from 'rollup-plugin-commonjs';
import uglify      from 'rollup-plugin-uglify';

//paths are relative to the execution path
export default {
  entry: 'src/main/resources/static/app/main-aot.js',
  dest: 'src/main/resources/static/dist/app.js', // output a single application bundle
  sourceMap: true,
  sourceMapFile: 'src/main/resources/static/dist/app.js.map',
  format: 'iife',
  onwarn: function(warning) {
    // Skip certain warnings

    // should intercept ... but doesn't in some rollup versions
    if ( warning.code === 'THIS_IS_UNDEFINED' ) { return; }

    // console.warn everything else
    console.warn( warning.message );
  },
  plugins: [
	  nodeResolve({
	    	jsnext: true, 
	        extensions: [ '.js', '.json' ],  // Default: ['.js']
	        module: true
	  }),
	  commonjs({
		  include: [
	    	  'node_modules/rxjs/**',
	    	  'node_modules/moment/**'
	    	  ]
	  }),
	  {
		  name: 'replace moment imports',
		  transform: code =>
          ({
        	  code: code.replace(/import\s*\*\s*as\s*moment/g, 'import moment'),
        	  map: { mappings: '' }
          })
      },
      uglify()
  ]
}
