// 3.0.0/src/main/resources/static/app/searchmodule/search.component.ts
import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';
import { Title } from '@angular/platform-browser';


import { SearchService } from "./search.service";
import { Customer } from "../shared/models/customer";





@Component({
  moduleId: module.id,
  selector: 'atlas-search',
  templateUrl: 'search.component.html',
  styleUrls: ['search.component.css']
})
  
export class SearchComponent implements OnInit{
    model: any = {};
    loading = true;
    returnUrl: string;
    customer: Customer[];
  
    constructor(
      private titleService: Title,
      private router: Router,
      private searchService: SearchService) {
        
        titleService.setTitle('ATLAS - Search');
    }

   
    ngOnInit() {
    }
    
     search() {
        this.loading = false;
        console.log(this.loading);
        this.searchService.search(this.model.refnum, this.model.reftype)
            .subscribe(
                customer => {
                	this.customer = customer;
                  	console.log(this.customer);
                },
                error => {
                    console.log(error);
                    this.loading = true;
                });
    }
  
}
